timesync/clock_test.go

```diff
@@ -19,34 +19,28 @@ func (MockTimer) Now() time.Time {
 	return start
 }
 
-func TestTicker_StartClock(t *testing.T) {
+func TestClock_StartClock(t *testing.T) {
 	tick := 1 * time.Second
-	layout := "2006-01-02T15:04:05.000Z"
-	str := "2018-11-12T11:45:26.371Z"
-	start, _ := time.Parse(layout, str)
-
-	ts := NewTicker(MockTimer{}, tick, start)
+	c := RealClock{}
+	ts := NewClock(c, tick, c.Now())
```

Comment:
I generally prefer tests that are static and repeatable, so the previous implementation (passing a specific time) is better, IMO.

If you insist on using the current wall time in the test, then at least log it so we can reproduce failures.

Reply:
Changed partially.
For timer_test.go it is true you can simply mock it so I changed it to use a mock.
For clock_test.go he measures actual duration (because it uses a real clock). Changing that to a mock will require that the mock will be able to advance by itself which is probably what we want.
Let me know if you think otherwise. I think it also means that the previous tests didn't really test it.